.github/workflows/opensearch-sink-os-integration-tests.yml

```diff
@@ -3,7 +3,17 @@
 
 name: Data Prepper OpenSearchSink integration tests with OpenSearch
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    paths:
+      - 'data-prepper-plugins/opensearch/**'
```

Comment:
How about adding path for data-prepper-core ? With conditional routing, the sink will dependent on some of the core packages ?

Reply:
I thought about that. But, this project doesn't actually use the sink pipelines and conditional routing. These tests only test the OpenSearchSink in isolation from Data Prepper core.

The following snippet is a good example of how the tests work. They only exercise calling methods directly on the sink:

https://github.com/opensearch-project/data-prepper/blob/23911dd95707750a8cf51888daf5b56685e4d2f4/data-prepper-plugins/opensearch/src/integrationTest/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java#L171-L173